plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XBuildManager.java

```diff
@@ -125,7 +123,7 @@ public String toString() {
 	private XWorkspaceManager workspaceManager;
 
 	@Inject
-	private Provider<TopologicalSorter> sorterProvider;
+	private XTopologicalSorter topoSorter;
```

Comment:
The sorter is stateful. We have to use a Provider to get a new instance per run.